suncal/models/astro.py

```diff
@@ -44,6 +46,25 @@ def rise_set_dict(
     return events
 
 
+def extract_moon_phase(
+    skyfield_t: Time, skyfield_y: np.ndarray, timezone: str
+) -> Tuple[str, str, dt.datetime]:
+    """
+    Using the information in skyfield_y we extract the name and the symbol of the moon phase that was found.
+    The Time provided in skyfield_t is converted to an aware datetime timestamp.
+    There is only one moon phase max per day - so we can safely assume that the skyfield_t and the skyfield_y array
+    hold only 1 item, respectively.
+    """
+
+    event_time = skyfield_t.astimezone(pytz.timezone(timezone)).item()
+
+    phase_idx = skyfield_y.item()
+    phase_symbol = MOON_PHASE_SYMBOLS[phase_idx]
+    phase_name = almanac.MOON_PHASES[phase_idx]
+
+    return phase_name, phase_symbol, event_time
```

Comment:
Why not return just `phase_idx` and leave the dict lookups to the caller?

Reply:
Because the `phase_idx` is quite cryptic. This way the cryptic part is abstracted away within the function.